modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -641,6 +650,38 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
         };
     }
 
+#ifdef HAVE_CUDA
+    Ptr<BackendNode> initCUDA(
+        void *context_,
+        const std::vector<Ptr<BackendWrapper>>& inputs,
+        const std::vector<Ptr<BackendWrapper>>& outputs
+    ) override
+    {
```

Comment:
> I don't know if there is some way to get the shape of the input in supportBackend(). 

You can do this check in `initCUDA`. Take a look at `CudaBackendWrapper`, it should have the host mat and you can get the shape of input from the host mat.

Reply:
I do use it, but `initCUDA` is called after `supportBackend` so it doesn't fallback to the cpu backend.